### DISCUSSION TITLE
simple where noop / conditional where clause?

### DISCUSSION THREAD

**Discussion by sa-0001** [jeremyevans/sequel#1883] [General]
hello, i'm wondering if there might be a simpler way to have a conditional `.where` in a chained query.  i have the following which works as expected, other than the unnecessary `1=1` being generated:

```
DB[:table_name]
    .where(...)
    .where(if types then { type: types } else Sequel.lit('1=1') end)
    .order(...)
    .limit(...)
```

if i use `.where(if X then X else nil end)`, this generates `AND NULL` into the query.  is there another possibility i may have missed?

**Comment by janko** [comment]
If you wanted to keep chaining, you can use `#yield_self` with a ternary operator:

```rb
DB[:table_name]
  .where(...)
  .yield_self { |ds| types ? ds.where(type: types) : ds }
  .order(...)
  .limit(...)
```

**Reply by sa-0001** [reply]
@janko thank you, that's what i was looking for.  this is much more flexible.  apparently this is not Sequel-specific but Ruby syntax?  my new fact of the day.
